File: r/R/list_datasets.R
#' List available datasets in the Bedrock Bio library
#'
#' @returns A character vector of dataset identifiers
#'
#' @examplesIf bedrockbio:::has_connection()
#' library(bedrockbio)
#' list_datasets()
#'
#' @export
list_datasets <- function() {
  catalog <- get_catalog()
  names(catalog)
}

File: python/src/bedrock_bio/list_datasets.py
from .utils import get_catalog


def list_datasets() -> list[str]:
    """
    List available datasets in the Bedrock Bio library.

    Returns
    -------
    list[str]
        A list of dataset identifiers (e.g. 'ukb_ppp.pqtls').

    Raises
    ------
    ConnectionError
        If the catalog cannot be accessed.

    Examples
    --------
    >>> import bedrock_bio as bb
    >>> bb.list_datasets()
    ['ukb_ppp.pqtls', ...]

    """
    catalog = get_catalog()
    return list(catalog.keys())
